### PR TITLE
handle source error

### PR DESCRIPTION
**Pull request by pacak**
Problem being solved:
type  error:
```
% cat Foo.hs
main = return ()
% hdevtools check Foo.hs
/home/pacak/hdevtools/Foo.hs:1:1: Warning:
    Top-level binding with no type signature: main :: IO ()
% hdevtools admin --status
Server is running.
```

source error:
```
% cat Bar.hs
import Perlude
main = return ()
% hdevtools check Bar.hs
hdevtools: <socket: 3>: hGetLine: end of file
% hdevtools admin --status
hdevtools: connect: does not exist (No such file or directory)
```

new version:
```
% hdevtools check Bar.hs
/home/pacak/hdevtools/Bar.hs:1:8:
    Could not find module ‘Perlude’
    Perhaps you meant Prelude (from base-4.8.0.0)
    Use -v to see a list of the files searched for.
% hdevtools admin --status
Server is running.
```

### CODE REVIEW

**Comment by ch1bo** [comment]
Can you make this fast-forward now?

**Comment by pacak** [comment]
> Can you make this fast-forward now?

Sure. Btw, #62 solves exactly the same problem but in a slightly different way - my tries to handle unknown exceptions if there's any, #62 tries to change less code.

**Comment by ch1bo** [comment]
Yeah I saw #62, and think handling the errors and communicating them through the socket is the cleaner solution. Maybe @takano-akio agrees?

**Comment by pacak** [comment]
> Also, you might want to remove the `Ghc.handleSourceError` and handler:

Looks like it, but I'm going to test it first. I'll poke Akio to show up here.

**Comment by pacak** [comment]
removed that extra handle, works for me.

**Comment by takano-akio** [comment]
> think handling the errors and communicating them through the socket is the cleaner solution.

Hmm, this seems to apply to the both patches. Which version are you referring to?

My rationale for handling source errors in `loadTarget` was that we were already handling source errros from the call to `GHC.load` there, and not catching errors from `depanal` seemed to be an accidental breakage introduced in 1d21b8d22. Other than that, I don't have preference one way or the other.

**Comment by takano-akio** [comment]
By the way, sorry for opening a duplicate pull request. I should have looked more carefully before submitting.

**Comment by ch1bo** [comment]
Seems like my reply via mail didn't end up here. Well

> Which version are you referring to?

Let me rephrase, sending the error to the frontend process - the one directly spawned via command line - from the backend process - the one which eventually was forked when the domain socket was created (the first run), is what I prefer. That way, the error always ends up in the process whose stdout/stderr is available to the user.

Also, I think I need to create a ticket for solving these Travis issues.

**Comment by takano-akio** [comment]
> Let me rephrase, sending the error to the frontend process - the one directly spawned via command line - from the backend process - the one which eventually was forked when the domain socket was created (the first run), is what I prefer. That way, the error always ends up in the process whose stdout/stderr is available to the user.

FWIW my patch (and the original code) did this as well. The error message was generated by `GHC.handleSourceError` and sent to the configurable `log_action` action in `DynFlags`. In hdevtools this action is set to `logAction`, which forwards error messages to the client using `clientSend`.
